test/spectral/tests/spectral-validator.test.js

```diff
@@ -108,3 +112,155 @@ describe('spectral - test spectral-validator.js', function() {
     );
   });
 });
+
+const sequential = process.argv.includes('--runInBand');
```

Comment:
My only question on the tests is - is there a way to just look at what rules spectral is ready to use after it's defined? Then we could just call the setup function and make sure it does what we want it to. I think that would be much simpler.

That said, I do like the idea of having an end-to-end test to verify the behavior we're looking for in a real life scenario. We can discuss later

Reply:
I was able to spyOn the config files so the end-to-end test isn't as troublesome anymore. For non-end-to-end tests, it's a bit difficult with spectral since creating a ruleset needs a file first and add/modifying rules is a bit messy using spectral's functions.